README.md
lliurex-ltsp
============

LTSP Customization for LliureX


lliurex-ltsp

Merging



debian/changelog
lliurex-ltsp (0.26) precise; urgency=high

  * Deleted flashplugin transfer from server, it will be packaged.

 -- Jose A. Murcia <joamuran@gmail.com>  Thu, 31 Oct 2013 11:36:07 +0100

lliurex-ltsp (0.25) precise; urgency=high

  * Added flashplugin transfer capabilities from server

 -- Jose A. Murcia <joamuran@gmail.com>  Thu, 31 Oct 2013 09:50:13 +0100

lliurex-ltsp (0.24) precise; urgency=high

  * Added new devilspie rules and openbox as a window manager into Xephyr Sessions.
  * Added umount for if the flies
  * Removed earlymirror

 -- Jose A. Murcia <joamuran@lite>  Thu, 31 Oct 2013 08:54:34 +0100

lliurex-ltsp (0.23) precise; urgency=low

  * Fixed bug in pxe menu creation (for laptops)

 -- Jose A. Murcia <joamuran@gmail.com>  Fri, 25 Oct 2013 12:18:25 +0200

lliurex-ltsp (0.22) precise; urgency=low

  * Getting mirror packages from file instead of http.

 -- Jose A. Murcia <joamuran@gmail.com>  Wed, 25 Sep 2013 12:18:04 +0200

lliurex-ltsp (0.21) precise; urgency=high

  * Postinstall fixed 

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Wed, 18 Sep 2013 08:40:14 +0200

lliurex-ltsp (0.20) precise; urgency=high

  * Postinstall execution of regeneration menu for PXE

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Tue, 17 Sep 2013 19:55:00 +0200

lliurex-ltsp (0.19) precise; urgency=low

  * Updated the logic:
    - Simple explanation: The label in a PXE entry is only
    a way to identify the entry at PXE menu. Is better that this
    entry only contains words like llx-client (for example) instead
    a "LliureX installation for the common people (LliureX 13.06)" label.
    
      

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Tue, 17 Sep 2013 19:43:57 +0200

lliurex-ltsp (0.18) precise; urgency=high

  * Updated default boot to llx-client

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Mon, 16 Sep 2013 11:35:08 +0200

lliurex-ltsp (0.17) precise; urgency=high

  * Solved LVDS bug that makes belive the thin client that had two vgas. 

 -- joamuran <joamuran@gmail.com>  Thu, 12 Sep 2013 14:04:51 +0200

lliurex-ltsp (0.16) precise; urgency=low

  * Changed default boot option

 -- David Montalva Furio <montalva_dav@gva.es>  Thu, 01 Aug 2013 09:50:47 +0200

lliurex-ltsp (0.15) precise; urgency=high

  * Updated with prerm abort-upgrade

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Thu, 18 Jul 2013 13:40:20 +0200

lliurex-ltsp (0.14) precise; urgency=low

  * Updated with prerm abort-upgrade

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Thu, 18 Jul 2013 13:39:05 +0200

lliurex-ltsp (0.13) precise; urgency=high

  * Fixed bug in i386 directory creation.

 -- Jose A. Múrcia <joamuran@gmail.com>  Wed, 17 Jul 2013 09:03:56 +0200

lliurex-ltsp (0.11) precise; urgency=high

  * Updated changelog.

 -- Jose A. Múrcia <joamuran@gmail.com>  Tue, 16 Jul 2013 14:02:18 +0200

lliurex-ltsp (0.10) precise; urgency=low

  * Updated PXE menu for boot from local disk

 -- root <root@gmail.com>  Tue, 16 Jul 2013 09:47:52 +0200

lliurex-ltsp (0.9) precise; urgency=high

  * Ignoring kernel versions in update kernels

 -- Jose A. Múrcia <joamuran@gmail.com>  Fri, 12 Jul 2013 14:35:15 +0200

lliurex-ltsp (0.8) precise; urgency=low

  * Added new images and enhaced PXE menu generation.

 -- Jose A. Múrcia <joamuran@gmail.com>  Thu, 11 Jul 2013 14:11:27 +0200

lliurex-ltsp (0.7) precise; urgency=high

  * Critical bug Fixed! link LliureX to Ubuntu.

 -- Jose A. Múrcia <joamuran@gmail.com>  Fri, 05 Jul 2013 13:20:39 +0200

lliurex-ltsp (0.6.7) precise; urgency=low

  * Updated template lts.conf

 -- Jose A. Múrcia <joamuran@gmail.com>  Wed, 26 Jun 2013 13:58:13 +0200

lliurex-ltsp (0.6.6) precise; urgency=low

  * Modified lliurex-ltsp-create-client to force server mirror.

 -- Jose A. Múrcia <joamuran@gmail.com>  Tue, 18 Jun 2013 13:09:11 +0200

lliurex-ltsp (0.6.5) precise; urgency=low

  * Modified lliurex-ltsp-create-client to force i386 arch and --purge-chroot

 -- Jose A. Múrcia <joamuran@gmail.com>  Tue, 30 Apr 2013 13:58:32 +0200

lliurex-ltsp (0.6.4) precise; urgency=low

  * Fixed /usr/sbin/lliurex-ltsp-create-client to include multiple packages into client with the initial install

 -- Jose A. Múrcia <joamuran@gmail.com>  Wed, 17 Apr 2013 10:00:57 +0200

lliurex-ltsp (0.6.3) precise; urgency=low

  * Updated to the new class model infrastructure: /etc/home -> /etc/server-sync/home

 -- Jose A. Múrcia <joamuran@pandora>  Mon, 15 Apr 2013 12:02:37 +0200

lliurex-ltsp (0.6.2) precise; urgency=low

  * Modified postinstall to keep /var/lib/tftpboot/ltsp/lts.conf if it exists

 -- Jose A. Múrcia <joamuran@gmail.com>  Mon, 15 Apr 2013 09:56:45 +0200

lliurex-ltsp (0.6.1) precise; urgency=low

  * Fixed path for lliurex-ltsp-create-client binary.
    Now is needed by the n4d-plugin of ltsp suite.

 -- Angel Berlanas Vicente <angel.berlanas@gmail.com>  Fri, 12 Apr 2013 11:44:59 +0200

lliurex-ltsp (0.6) precise; urgency=low

  * Integration with dnsmasq

 -- Jose A. Múrcia <joamuran@gmail.com>  Tue, 26 Feb 2013 13:58:53 +0100

lliurex-ltsp (0.5) precise; urgency=low

  * Added epoptes as dependency to install server keys into server in plugins/ltsp-build-client/common/040-epoptes-certificate

 -- Jose A. Múrcia <joamuran@pandora>  Fri, 22 Feb 2013 10:17:31 +0100

lliurex-ltsp (0.4) precise; urgency=low

  * Fixed diversions well - I hope...

 -- Jose A. Múrcia <joamuran@gmail.com>  Wed, 13 Feb 2013 12:50:03 +0100

lliurex-ltsp (0.3) precise; urgency=low

  * Fixed diversions

 -- Jose A. Múrcia <joamuran@gmail.com>  Fri, 08 Feb 2013 10:19:28 +0100

lliurex-ltsp (0.2) precise; urgency=low

  * Fixed scripts X01-localapps and X99-zlocalapps-cleanup to mount homes

 -- Jose Alfredo Murcia Andres <joamuran@gmail.com>  Tue, 04 Dec 2012 12:00:08 +0100

lliurex-ltsp (0.1) UNRELEASED; urgency=low

  * Initial release.

 -- joamuran <joamuran@gmail.com>  Fri, 09 Nov 2012 11:48:22 +0100


install/usr/share/lliurex-ltsp/q3llum.sh
#!/bin/bash
mk_users_passwd() {
    user=$1
    pass=$2
    # ¡Magia, no toques aqui!
    /usr/sbin/chpasswd <<END
$user:$pass
END
}

mk_net_structure(){
	username=$1
	group=$2

 	mkdir -p /net/server-sync/home/${group}/${username}/UserFiles

       	for folder in Documents Groups Music Videos Pictures Share; do
       		mkdir -p /net/server-sync/home/${group}/${username}/UserFiles/${folder}
       	done

       chown -R ${username}:${group} /net/server-sync/home/${group}/${username}



}
 
# USAGE: q3llum.sh group_name group_prefix initusernumber endusernumber
 
group=$1
prefix=$2

for  (( i = $3 ; i <= $4; i++ )); do
        if [ $i -lt 10 ]; then
                username=${prefix}0${i}
        else
                username=${prefix}${i}
        fi

	echo "Creating user=${username} password=${username} group=${group}"
        /usr/sbin/useradd --home /home/$username --create-home --gid $group $username || exit 1

        #Create password
	mk_users_passwd ${username} ${username} || exit 1

	#Create structure in /net
	mk_net_structure ${username} ${group}

	# modify user shell
	chsh -s /bin/bash ${username}

	# Add user to fuse group
	adduser ${username} fuse
done


install/usr/share/lliurex-ltsp/llx-cp-pxe.sh
#!/bin/bash
# This command makes a copy of pxelinux.cfg/default files to be booted on a multiple LliureX LTSP images system

init() {
    default=$1
    destination=$2

    # Initial conditions, usage, etc.
if [ $# -eq 4 ];then
    option="interactive"
else
    option=$3
fi

if [ $1 = $2 ] && [ $option != "force" ]; then
        read -p "Warning! Destination and Original file are the same! Are you sure? " -n 1
    if [[ ! $REPLY =~ ^[Yy]$ ]]
    then
        exit 1
    fi    
fi
}

# Initial conditions, usage, etc.
if [ $# -ne 5 ] && [ $# -ne 4 ]; then
    echo "Usage: llx-cp-pxe filename-to-magic-copy destination-to-copy-the-llx-pxe-magic [interactive|force] ip-server chroot-name"
    exit -1
fi

# getting parameters
default=$1
destination=$2
option=$3
server=$4
chroot=$5

init ${default} ${destination} ${option};


# Let's go!

echo "# File created by LliureX LTSP command llx-cp-pxe" > "/tmp/llx-cp-pxe.tmp"

echo "default ltsp-NBD" >> "/tmp/llx-cp-pxe.tmp"
echo "ontimeout ltsp-NBD" >> "/tmp/llx-cp-pxe.tmp"
echo "" >> "/tmp/llx-cp-pxe.tmp"

adding_ltsp=0
while read line
  do
      
   if [ "$line" = 'label ltsp' ]; then
        line='label ltsp-NBD'
   fi

   if [ "$line" = 'label ltsp-NBD' ]; then
      adding_ltsp=1
   fi

    # Transform menu lines
    header=$(echo $line | cut -d " " -f1)
    #
    #if [ "$header" = 'ipappend' ] && [ $adding_ltsp -eq 1 ]; then
    #    adding_ltsp=0
    #    echo "ipappend 3" >> /tmp/llx-cp-pxe.tmp
    #fi
    
    if [ "$header" = 'append' ] && [ $adding_ltsp -eq 1 ]; then
        line="$line nbdroot=$server:$chroot video=LVDS-1:d"
    fi
        
    if [ "$line" = 'ipappend 2' ]; then
        line=""
    fi


    if [ $adding_ltsp -eq 1 ] && [ "$header" != 'ipappend' ]; then
	ISINITRDLINE=`echo $line | grep initrd | wc -l`
	ISKERNELLINE=`echo $line | grep vmlinuz | wc -l`

	# parsin initrd.imgxxxx
	if [ $ISINITRDLINE -eq 1 ]; then
		SUSTITUYE="$(echo $line|cut -d "=" -f2| cut -d " " -f1)"
		line=$(echo $line | sed "s%$SUSTITUYE%initrd.img%g")
	fi

	# parsin vmlinux...
	if [ $ISKERNELLINE -eq 1 ]; then
		SUSTITUYE="$(echo $line|cut -d " " -f2)"
		line=$(echo $line | sed "s%$SUSTITUYE%vmlinuz%g")
	fi


        echo $line >> /tmp/llx-cp-pxe.tmp
    fi
    if [ -z "$line" ]; then
        adding_ltsp=0
    fi

done < $default

echo "ipappend 3" >> /tmp/llx-cp-pxe.tmp

mv /tmp/llx-cp-pxe.tmp $destination


install/usr/share/lliurex-ltsp/llx-create-pxelinux.sh
#!/bin/bash
# This command regenerates PXELinux Menus


getlabel(){
    dir=$1
    name=$(echo $dir | cut -d "/" -f1)
    menulabel=$name

    case "$name" in
    "llx-desktop")
        menulabel="LliureX 13.06 Escriptori"
        ;;
    "llx-client")
        menulabel="LliureX 13.06 Client d'aula"
        ;;
    "llx-infantil")
        menulabel="LliureX 13.06 Infantil"
        ;;
    "llx-musica") 
        menulabel="LliureX 13.06 Musica"
        ;;
    "llx-pime")
        menulabel="LliureX 13.06 Pime"
        ;;
    "llx-lite")
        menulabel="LliureX 13.06 Escriptori Lleuger (XFCE)"
        ;;
    esac
    echo $menulabel;    
}


tftpboot="/var/lib/tftpboot/ltsp"

# Check if pxelinux exists
if [ -f "$tftpboot/pxelinux.cfg/default" ]; then
	# Check if netinstall is available
	NETINST=`cat $tftpboot/pxelinux.cfg/default | grep "# Netinst: Install Menu" | wc -l`    
else 
	# if not exists pxelinux.cfg/default, let's allow netinstall
	NETINST=1
fi


echo "#LliureX LTSP Boot Menu - Created automatically by llx-create-pxelinux
DEFAULT pxelinux.cfg/vesamenu.c32
PROMPT 0
MENU TITLE Arrencada per xarxa de LliureX
MENU BACKGROUND lliurex-pxe.png

MENU WIDTH 80
MENU MARGIN 10
MENU ROWS 12
MENU TABMSGROW 18
MENU CMDLINEROW 12
MENU ENDROW 24
MENU TIMEOUTROW 20
menu color title 2 #ff000000 #00999999 std
menu color unsel 0 #ff000000 #00999999 std
menu color sel 30,47 #00990099 #889900cc std
menu color border 0 #00ffffff #00ffffff none
TIMEOUT 50

LABEL localboot
      menu label ^Arranca des del disc dur
      localboot 0
" > $tftpboot/pxelinux.cfg/default 

for dir in `ls $tftpboot`
do
    if [ -d $tftpboot/$dir ] && [ -e "$tftpboot/$dir/pxelinux.cfg/default" ] && [ $dir != "netinstall" ]; then

        #echo "$dir/pxelinux.cfg/default"
	# This logic is here to prevent the mistake
	# that LABEL is the GOTO name
	
	label_of_label=$(echo $dir | cut -d "/" -f1)
	
	# The label value is the Human readable
	# value for this PXE entry.
        
	label=$(getlabel ${dir})

	# This is the dir of pxe linux (submenus)
	
        dirpxe=$(echo $dir | cut -d "/" -f1)
	
	# Creating the pxe entry at menu
        echo "llx-create-pxelinux: Adding menu for $label"
	echo "LABEL $label_of_label
    MENU LABEL Arranca $label" >> $tftpboot/pxelinux.cfg/default
	
	# This is the DEFAULT PXE entry
	# is important to be setted to llx-client
	if [ "$label_of_label" = "llx-client" ]; then
		echo "    MENU default" >> $tftpboot/pxelinux.cfg/default
	fi
	
	# Continue with the rest of entry
 	echo "KERNEL pxelinux.cfg/vesamenu.c32
    CONFIG $dir/pxelinux.cfg/default $dirpxe/">> $tftpboot/pxelinux.cfg/default


	#### BEGIN ENABLE LAPTOPS MENU
	rc=0
	#cat $tftpboot/$dir/pxelinux.cfg/default | grep -q 'video=LVDS-1:d' || rc=1
	#if [ ${rc} -eq 1 ]; then 

	if [ `cat $tftpboot/$dir/pxelinux.cfg/default | grep 'video=LVDS-1:d' | wc -l` -eq 1 ]; then
		cat $tftpboot/$dir/pxelinux.cfg/default | sed 's/video=LVDS-1:d//' > $tftpboot/$dir/pxelinux.cfg/default-laptop
		# Creating the pxe entry at menu
        	echo "llx-create-pxelinux: Adding menu for $label (laptop)"
		echo "LABEL $label_of_label-laptop
	MENU LABEL Arranca $label (Ordinador Portatil)" >> $tftpboot/pxelinux.cfg/default
		
		# Continue with the rest of entry
	 	echo "KERNEL pxelinux.cfg/vesamenu.c32
	CONFIG $dir/pxelinux.cfg/default-laptop $dirpxe/">> $tftpboot/pxelinux.cfg/default
	
	fi
	#### END ENABLE LAPTOPS MENU

    fi
done


if [ $NETINST -eq 1 ]; then
        echo "llx-create-pxelinux: Adding menu Netinstall"
	cat << EOF >> $tftpboot/pxelinux.cfg/default

# Netinst: Install Menu 
LABEL Instal.la LliureX en aquest ordinador
   MENU LABEL Instal.la LliureX en aquest ordinador
   KERNEL netinstall/ubuntu-installer/i386/boot-screens/vesamenu.c32
   CONFIG netinstall/pxelinux.cfg/default netinstall/
EOF

fi
